lib/elixir/lib/macro/env.ex

```diff
@@ -561,6 +562,12 @@ defmodule Macro.Env do
     * `:check_deprecations` - when set to `false`, does not check for deprecations
       when expanding macros
 
+    * `:local_for_callback` - a function that receives the metadata, name, arity,
```

Comment:
Let's not add a new option, let's augment `allow_locals` to be a function? Then we don't need to worry about how these two different APIs compose.